Geometry_Engine/Modify/FilletPolyline.cs

```diff
@@ -0,0 +1,164 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2025, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.Engine.Geometry;
+using BH.oM.Base.Attributes;
+using BH.oM.Geometry;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Net.NetworkInformation;
+
+namespace BH.Engine.Geometry
+{
+    public static partial class Modify
+    {
+        /***************************************************/
+        /**** Public Methods - Curves                  ****/
+        /***************************************************/
+
+        [Description("Returns a PolyCurve representing the input Polyline with fillets of the given radius at each internal vertex.")]
+        [Input("polyline", "The Polyline to fillet.")]
+        [Input("radius", "The radius of the fillet arc at each corner.")]
+        [Output("polyCurve", "A PolyCurve with fillets at each corner, or null if unsuccessful.")]
+
+        public static PolyCurve Fillet(this Polyline polyline, double radius)
+        {
+            if (polyline == null || polyline.ControlPoints == null || polyline.ControlPoints.Count < 3)
+            {
+                Base.Compute.RecordError("incorrect polyline for filleting");
```

Comment:
This error message needs to be more descriptive, have proper punctuation and SentenceCase for types e.g. Polyline.